src/main/java/services/petshop/steps/CommonMethodsPetSteps.java

```diff
@@ -7,34 +7,41 @@
 import net.thucydides.core.annotations.Step;
 import net.thucydides.core.annotations.Steps;
 import services.petshop.pojo.requests.Pet;
+import utils.methods.ReusableMethods;
 
 public class CommonMethodsPetSteps {
 
+    ReusableMethods reusableMethods = new ReusableMethods();
+
     @Steps
     PostPetSteps postPetSteps;
 
     @Steps
     GetPetSteps getPetSteps;
 
-    @Step("When I retrieve the id of the created pet")
-    public void getTheIdOfTheCreatedPet(Pet mypet){
+    @Step("When I create a new pet and retrieve the id of the created pet")
+    public int createPetAndGetItsId(Pet mypet) {
         Response createPetR = postPetSteps.createNewPet(mypet);
         Pet createdPet = createPetR.as(Pet.class);
-        System.out.println(getPetSteps.getPetWithId(createdPet.getId()));
+        return createdPet.getId();
     }
 
-    @Step("Then I check the status code of the response when I perform an action successfully is 200")
-    public void statusCodeSuccess(Response response){
-        Assert.assertEquals("The status is not 200 OK.",200, response.getStatusCode());
+    @Step("Then I check the status code of the response when I perform this action is {1}")
+    public void statusCode(Response response, int statusCode) {
+        reusableMethods.validateResponseStatusCode(response, statusCode);
     }
```

Comment:
you created here an unnecessary wrapper. Instead of creating 'statusCode' you can just use reusableMethods.validateResponseStatusCode